htmresearch/frameworks/utils/param_finder.py

```diff
@@ -0,0 +1,380 @@
+        effects are still visible.
+
+      'valid':
+        Mode `valid` returns output of length
+        ``max(M, N) - min(M, N) + 1``.  The convolution product is only given
+        for points where the signals overlap completely.  Values outside
+        the signal boundary have no effect.
+
+  Returns
+  -------
+  out : ndarray
+      Discrete, linear convolution of `a` and `v`.
+
+  References
+  ----------
+  .. [1] Wikipedia, "Convolution", http://en.wikipedia.org/wiki/Convolution.
+
+  """
+  a, v = np.array(a, ndmin=1), np.array(v, ndmin=1)
+  if len(v) > len(a):
+    a, v = v, a
+  if len(a) == 0:
+    raise ValueError('a cannot be empty')
+  if len(v) == 0:
+    raise ValueError('v cannot be empty')
+  mode = _mode_from_name(mode)
+  return np.core.multiarray.correlate(a, v[::-1], mode)
+
+
+
+def _mode_from_name(mode):
+  if isinstance(mode, basestring):
+    return _mode_from_name_dict[mode.lower()[0]]
+  return mode
+
+
+
+def _ricker_wavelet(points, a):
+  """
+  Return a Ricker wavelet, also known as the "Mexican hat wavelet".
+
+  It models the function:
+
+      ``A (1 - x^2/a^2) exp(-t^2/a^2)``,
+
+  where ``A = 2/sqrt(3a)pi^1/3``.
+
+  Parameters
+  ----------
+  points : int
+      Number of points in `vector`.
+      Will be centered around 0.
+  a : scalar
+      Width parameter of the wavelet.
+
+  Returns
+  -------
+  vector : (N,) ndarray
+      Array of length `points` in shape of ricker curve.
+
+  """
+  A = 2 / (np.sqrt(3 * a) * (np.pi ** 0.25))
+  wsq = a ** 2
+  vec = np.arange(0, points) - (points - 1.0) / 2
+  tsq = vec ** 2
+  mod = (1 - tsq / wsq)
+  gauss = np.exp(-tsq / (2 * wsq))
+  total = A * mod * gauss
+  return total
+
+
+
+def _cwt(data, wavelet, widths):
+  """
+  Continuous wavelet transform.
+
+  Performs a continuous wavelet transform on `data`,
+  using the `wavelet` function. A CWT performs a convolution
+  with `data` using the `wavelet` function, which is characterized
+  by a width parameter and length parameter.
+
+  Parameters
+  ----------
+  data : (N,) ndarray
+      data on which to perform the transform.
+  wavelet : function
+      Wavelet function, which should take 2 arguments.
+      The first argument is the number of points that the returned vector
+      will have (len(wavelet(width,length)) == length).
+      The second is a width parameter, defining the size of the wavelet
+      (e.g. standard deviation of a gaussian). See `ricker`, which
+      satisfies these requirements.
+  widths : (M,) sequence
+      Widths to use for transform.
+
+  Returns
+  -------
+  cwt: (M, N) ndarray
+      Will have shape of (len(data), len(widths)).
+
+  """
+  output = np.zeros([len(widths), len(data)])
+  for ind, width in enumerate(widths):
+    wavelet_data = wavelet(min(10 * width, len(data)), width)
+    output[ind, :] = _convolve(data, wavelet_data,
+                               mode='same')
+  return output
+
+
+
+def read_csv_files(fileName):
```

Comment:
We may or may not. The param finder will need to read data from the file system in Unicorn. Also, the actual script in unicorn will most likely be structured differently / moved somewhere else, so it's probably ok to leave this script like this for now. What are your thoughts?